src/auth/auth.service.ts

```diff
@@ -59,6 +61,34 @@ export class AuthService {
     };
   }
 
+  async loginSteam(loginDto: CreateSteamUserDto) {
+    let user = await this.steamUserService.findOneBySteamId(loginDto.steamid);
+
+    if (!user) {
+      user = await this.steamUserService.create(loginDto);
+    }
+
+    const tokens = await this.issueTokens(user.id);
+
+    return {
+      user,
+      ...tokens,
+      jwtExpiresIn: getTime(addSeconds(new Date(), JWT_EXPIRES_IN_SECONDS)),
+    };
+  }
```

Comment:
This function should find a `User` by the steamId. 
You can add the `findOne` method in `UserService`:
```ts
async findOne(where: Prisma.UserWhereInput) {
  return this.prisma.user.findFirstOrThrow({ where });
}
```
So later you can use it in `loginSteam` function:
```ts
async loginSteam(loginDto: CreateSteamUserDto) {
  const user = await this.userService.findOne({
    steamUser: { steamid: loginDto.steamid },
  });

  const tokens = await this.issueTokens(user.id);

  return {
    user,
    ...tokens,
    jwtExpiresIn: getTime(addSeconds(new Date(), JWT_EXPIRES_IN_SECONDS)),
  };
}
```

That'll allow you to use the same user model in any login strategy